vlm/src/main/scala/geotrellis/contrib/vlm/gdal/URIFormatter.scala

```diff
@@ -0,0 +1,91 @@
+package geotrellis.contrib.vlm.gdal
+
+
+import java.net.URI
+
+
+object Schemes {
+  final val FTP = "ftp"
+  final val HTTP = "http"
+  final val HTTPS = "https"
+
+  final val TAR = "tar"
+  final val ZIP = "zip"
+  final val GZIP = "gzip"
+
+  final val FILE = "file"
+
+  final val S3 = "s3"
+
+  final val GS = "gs"
+
+  final val WASB = "wasb"
+  final val WASBS = "wasbs"
+
+  final val HDFS = "hdfs"
+
+  final val FILE_TYPE_TO_SCHEME =
+    Map(
+      TAR -> TAR,
+      "tgz" -> TAR,
+      ZIP -> ZIP,
+      "kmz" -> ZIP,
+      "ods" -> ZIP,
+      "xlsx" -> ZIP,
+      GZIP -> GZIP
+    )
+}
+
+
+object URIFormatter {
```

Comment:
Can i quibble about the name a bit? ‘URIformatter’ could mean almost anything.

Some alternatives are: ‘UriVsiFormatter’, package function named ‘uri2vsi’ or ‘VsiPath’ that has ‘fromUri’ and ‘toUri’ methods.

Reply:
My vote would be for `VSIPath`, and we can wrap `URI` into it (smth like what was implemented in rasterio)